### PR TITLE
Use jwt block for signing

### PR DESCRIPTION
**Pull request by johakoch**
In order to simplify local JWT handling (create/sign and check), it should be possible to also reference `jwt` blocks from the `jwt_sign()` function.

To do this, we need
* `signing_ttl` and
* for RSA keys `signing_key`/`signing_key_file`.

So, for JWT signed with an `HS256` key
```
jwt_signing_profile "local_jwt" {
  signature_algorithm = "HS256"
  key = env.LOCAL_JWT_KEY
  ttl = "1h"
}
jwt "local_jwt" {
  signature_algorithm = "HS256"
  key = env.LOCAL_JWT_KEY
  cookie = "local_token"
}
```
could be simplified to

```
jwt "local_jwt" {
  signature_algorithm = "HS256"
  key = env.LOCAL_JWT_KEY
  cookie = "local_token"
  signing_ttl = "1h"
}
```

### CODE REVIEW

File: docs/REFERENCE.md [hunk 1]
```diff
@@ -346,10 +346,20 @@ required _label_.
...
 | `key`           |string|-|Public key (in PEM format) for `RS*` variants or the secret for `HS*` algorithm.|-|-|
 | `key_file`          |string|-|Optional file reference instead of `key` usage.|-|-|
-|  `signature_algorithm`           |string|-|-|&#9888; required. Valid values are: `RS256` `RS384` `RS512` `HS256` `HS384` `HS512`.|-|
+| `signature_algorithm`           |string|-|-|&#9888; required. Valid values are: `RS256` `RS384` `RS512` `HS256` `HS384` `HS512`.|-|
 | `claims`               |string|-|Equals/in comparison with JWT payload.|-|-|
 | `required_claims`      | string|-|list of claims that must be given for a valid token |-|-|
 
+The `jwt` block may also be referenced by the [`jwt_sign()` function](#functions), if it has a `signing_ttl` defined. For `HS*` algorithms the signing key is taken from `key`/`key_file`, for `RS*` algorithms, `signing_key` or `signing_key_file` have to be specified.
+
+*Note:* A `jwt` block with `signing_ttl` cannot have the same label as a `jwt_signing_profile` block.
```
Comment (malud) [inline docs/REFERENCE.md:355]:
```suggestion
> *Note:* A `jwt` block with `signing_ttl` cannot have the same label as a `jwt_signing_profile` block.
```

Comment (johakoch) [reply docs/REFERENCE.md:355]: `>` means block quote. This is *not* a block quote.

File: docs/REFERENCE.md [hunk 2]
```diff
@@ -346,10 +346,20 @@ required _label_.
...
+
+| Attribute(s) | Type |Default|Description|Characteristic(s)| Example|
+| :-------- | :--------------- | :--------------- | :--------------- | :--------------- | :--------------- |
+| `signing_key`       |string|-|Private key (in PEM format) for `RS*` variants.|-|-|
+| `signing_key_file`  |string|-|Optional file reference instead of `signing_key` usage.|-|-|
+| `signing_ttl`       |string|-|The token's time-to-live (creates the `exp` claim).|-|-|
```
Comment (alex-schneider) [inline docs/REFERENCE.md:361]: Type `duration`, see line 294

File: docs/REFERENCE.md (hunk 1 above)
Comment (alex-schneider) [inline docs/REFERENCE.md:355]: This condition is not checked in the `config/configload/load.go`

File: eval/lib/jwt.go [hunk 3]
```diff
@@ -13,30 +13,123 @@ import (
...
+		dur      time.Duration
+		parseErr error
+	)
+
+	alg := acjwt.NewAlgorithm(signatureAlgorithm)
+	if alg == acjwt.AlgorithmUnknown {
+		return dur, alg, fmt.Errorf("algorithm is not supported")
+	}
+
+	if ttl != "0" {
```
Comment (alex-schneider) [inline eval/lib/jwt.go:45]: ttl is of type `duration`. `0` is an invalid value. `0s`, `0ms`, ... is valid.

File: eval/lib/jwt.go [hunk 4]
```diff
@@ -13,30 +13,123 @@ import (
...
+	if j.SigningTTL == "" {
+		return nil, nil
+	}
+
+	ttl, alg, err := checkData(j.SigningTTL, j.SignatureAlgorithm)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
```
Comment (alex-schneider) [inline eval/lib/jwt.go:101]: why `(` and `)`? :)

File: eval/lib/jwt.go (hunk 3 above)
Comment (johakoch) [reply eval/lib/jwt.go:45]: This was implemented a long time ago in order to facilitate non-expiring tokens. That's why it was not documented as `duration` but as `string`.

File: docs/REFERENCE.md (hunk 1 above)
Comment (johakoch) [reply docs/REFERENCE.md:355]: https://github.com/avenga/couper/pull/309/files?file-filters%5B%5D=.go#diff-1747ea6352257c16eb33296a5382c7a112e4c1a18ccd292f20fa8fa9b7be8dc9R311

File: eval/lib/jwt.go (hunk 4 above)
Comment (johakoch) [reply eval/lib/jwt.go]: fixed

File: docs/REFERENCE.md (hunk 1 above)
Comment (alex-schneider) [reply docs/REFERENCE.md:355]: I mean `A jwt block with signing_ttl`. Here is no check for the `signing_ttl`.

**Review by alex-schneider — APPROVED** [review]

**Pull request merged**